### PR TITLE
feat(components): added HoursStatus component

### PR DESCRIPTION
**Pull request by Jeffrey-Rhoads17**
J=CR-2815
TEST=manual

### CODE REVIEW

File: packages/pages/src/components/hoursStatus/hoursStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+ * @param {Function} futureTemplate override rendering for the "future" part of this component "Open Now - [[closes at]] 5:00PM Monday"
+ * @param {Function} timeTemplate override rendering for the "time" part of this component "Open Now - closes at [[5:00PM]] Monday"
+ * @param {Function} dayOfWeekTemplate override rendering for the "dayOfWeek" part of this component "Open Now - closes at 5:00PM [[Monday]]"
+ */
+const HoursStatus: React.FC<HoursStatusProps> = (props) => {
+  const [hasStatusTimeout, setHasStatusTimeout] = useState(false);
+
+  // Use two rendering passes to avoid SSR issues where server & client rendered content are different
+  // https://reactjs.org/docs/react-dom.html#hydrate
+  const [isClient, setIsClient] = useState(false);
```
Comment (mkilpatrick) [inline packages/pages/src/components/hoursStatus/hoursStatus.tsx:126]: We have a util to check for client. Could that work here instead?

File: packages/pages/src/components/hoursStatus/hoursStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+export interface StatusParams {
+  isOpen: boolean;
+  currentInterval: HoursIntervalManipulator | null;
+  futureInterval: HoursIntervalManipulator | null;
+  timeOptions?: Intl.DateTimeFormatOptions;
+  dayOptions?: Intl.DateTimeFormatOptions;
+}
+
+export interface TemplateParams {
+  currentTemplate?: (s: StatusParams) => React.ReactNode;
```
Comment (mkilpatrick) [inline packages/pages/src/components/hoursStatus/hoursStatus.tsx:18]: I'm not sure how others feel but all of these props being call fooTemplate sounds weird to me.

File: packages/pages/src/components/hoursStatus/hoursStatus.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  timeOptions?: Intl.DateTimeFormatOptions;
+  dayOptions?: Intl.DateTimeFormatOptions;
+  statusTemplate?: (s: StatusParams) => React.ReactNode;
+  className?: string;
+}
+
+function defaultCurrentTemplate(params: StatusParams): React.ReactNode {
+  return (
+    <span className="HoursStatus-current">
+      {params.isOpen ? "Open Now" : "Closed"}
```
Comment (mkilpatrick) [inline packages/pages/src/components/hoursStatus/hoursStatus.tsx:38]: Should the defaults handle i18N automatically?

Comment (Jeffrey-Rhoads17) [reply packages/pages/src/components/hoursStatus/hoursStatus.tsx:38]: We're going to add this in later!

File: packages/pages/src/components/hoursStatus/hoursStatus.tsx (hunk 1 above)
Comment (Jeffrey-Rhoads17) [reply packages/pages/src/components/hoursStatus/hoursStatus.tsx:126]: Renamed variable to be a better representation of what it is doing

File: packages/pages/src/components/hoursStatus/hoursStatus.tsx (hunk 2 above)
Comment (bhainesva) [inline packages/pages/src/components/hoursStatus/hoursStatus.tsx:18]: One thing that could be a nice convenience is having these be able to take raw strings for cases where you don't care about the StatusParams and always want the same thing

**Comment by mkilpatrick** [comment]
Closing this out since all components have moved to https://github.com/yext/js. Please open a PR there if you'd still like to contribute this.

**Pull request closed without merging**